src/components/structures/RoomView.tsx

```diff
@@ -809,7 +810,21 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
         return this.state.room?.roomId ?? this.state.roomId;
     };
 
-    private getPermalinkCreatorForRoom(room: Room): RoomPermalinkCreator {
+    private getPermalinkCreatorForRoom(): RoomPermalinkCreator {
+        const { room, roomId } = this.state;
+
+        // if room is undefined, attempt to use the roomId to create and store a permalinkCreator
+        if (room === undefined) {
+            if (isNotUndefined(roomId)) {
+                const permalinkCreator = new RoomPermalinkCreator(null, roomId);
+                this.permalinkCreators[roomId] = permalinkCreator;
+                return permalinkCreator;
+            } else {
+                // this will throw every time, should we throw here instead?
+                return new RoomPermalinkCreator(null, null);
```

Comment:
I'm not sure about this addition. On the one hand I feel it makes sense, because we're trying to get a permalink creator and the throw will come from the `RoomPermalinkCreator`. On the other hand, it seems off to call when we know the result will be a throw.

Reply:
I'm surprised the signature is loose enough to allow this :D I think throwing an explicit error would be clearer

Reply:
Ok I'm glad you said that because that was my gut reaction as well. I'll give this a bit more time this morning to see what I come up with.

Reply:
Have amended this to throw from `RoomView` now